nf_core/lint/actions_schema_validation.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env python
+
+import logging
+import yaml
+import json
+import jsonschema
+import os
+import requests
+
+
+def actions_schema_validation(self):
+    """
+    Validate workflows against a schema
+
+    Uses the JSON Schema from
+    https://json.schemastore.org/github-workflow
+    to verify that all workflow yml files adhere to the correct Schema
+    """
+    passed = []
+    warned = []
+    failed = []
+
+    # Only show error messages from schema
+    logging.getLogger("nf_core.schema").setLevel(logging.ERROR)
+
+    # Get all workflow files
+    action_workflows = os.listdir(os.path.join(self.wf_path, ".github/workflows"))
+
+    # Load the GitHub workflow schema
+    r = requests.get("https://json.schemastore.org/github-workflow", allow_redirects=True)
+    schema = r.json()
+
+    # Validate all workflows against the schema
+    for wf in action_workflows:
+        # load workflow
+        wf_path = os.path.join(self.wf_path, ".github/workflows", wf)
+        with open(wf_path, "r") as fh:
+            wf_json = yaml.safe_load(fh)
```

Comment:
Would be good to catch errors raised from yaml syntax problems and log as a failure here.

Reply:
Good point - those files are also loaded in the other actions_* lint checks, I guess it makes sense to catch those errors there as well? Could basically just copy-paste the same code. We'll have a few redundant errors reported then, if the yml is wrong.